elastic-agent-shipper.yml

```diff
@@ -7,4 +7,16 @@ tls: false
 #log level
 logging.level: debug
 logging.selectors: ["*"]
-logging.to_stderr: true
\ No newline at end of file
+logging.to_stderr: true
+
+queue:
+  test: #There is no actual "test" queue type, remove this later.
+    events: 512
+  metrics:
```

Comment:
I'm not sure we want this to be a property of the queue. It should probably be a top level level configuration.

Also consider that users are unlikely to be configuring the shipper, they will be configuration the agent (likely with the output section being translated here).

We want the configuration applied to the shipper to come directly from the agent policy without any transformation. How does the agent enable the HTTP expvar/metrics endpoint? Does it allow that? Should this instead be collected using the diagnostic command interface and the command protocol.

I realize this is for initial development but we shouldn't let this diverge too much from what the final system should do.

Reply:
I was actually really on the fence about where the config should "live" but that makes sense, yah.

Reply:
Does the agent use the same configuration library as beats does? We may want to use the same configuration package the agent does if it doesn't to minimize any possibility for incompatibility.